### ISSUE TITLE
Provide documentation for BatchTracer when using pgxpool.Pool

### ISSUE THREAD

**Issue by rfloriano** [jackc/pgx#1347]
Hey!

I'm using `v5`, and trying to add some traces for queries executing using `pgxpool.Pool`, and I'm also using `pgx.Batch`. 
That said, I only found Tracer usage, like this:
```go
    config, err := pgxpool.ParseConfig(dbUrl)
    config.ConnConfig.Tracer = NewTracerQueryHook()
```

But I didn't find any way to use BatchTracer for tracing batches. I've tried the docs, the tests, and the code itself. But it's not clear to me, how to use that.
Can you provide some documentation on that?

Btw, thanks for this amazing library.

**Comment by rfloriano** [comment]
Found it! Closing the issue.

**TL;DR**
Just implementing the `BatchTracer` methods to the `Tracer` will work well.

**Rational**
[ConnConfig](https://github.com/jackc/pgx/blob/master/conn.go#L23) expect to the `Tracer` being a `QueryTracer`.

```go
type ConnConfig struct {
	pgconn.Config

	Tracer QueryTracer
        [...]
}
```

And the [QueryTracer](https://github.com/jackc/pgx/blob/master/tracer.go#L10) just want `TraceQueryStart` and `TraceQueryEnd` methods.
```go
// QueryTracer traces Query, QueryRow, and Exec.
type QueryTracer interface {
	// TraceQueryStart is called at the beginning of Query, QueryRow, and Exec calls. The returned context is used for the
	// rest of the call and will be passed to TraceQueryEnd.
	TraceQueryStart(ctx context.Context, conn *Conn, data TraceQueryStartData) context.Context

	TraceQueryEnd(ctx context.Context, conn *Conn, data TraceQueryEndData)
}
```

But, in the [connect](https://github.com/jackc/pgx/blob/master/conn.go#L195) function, the pgx tries to assert the `queryTracer` other types too. like this:
```go
	if t, ok := c.queryTracer.(BatchTracer); ok {
		c.batchTracer = t
	}
	if t, ok := c.queryTracer.(CopyFromTracer); ok {
		c.copyFromTracer = t
	}
	if t, ok := c.queryTracer.(PrepareTracer); ok {
		c.prepareTracer = t
	}
```

So, if my tracer implements the methods from [BatchTracer](https://github.com/jackc/pgx/blob/master/tracer.go#L29), it will work properly.

```go
type BatchTracer interface {
	// TraceBatchStart is called at the beginning of SendBatch calls. The returned context is used for the
	// rest of the call and will be passed to TraceBatchQuery and TraceBatchEnd.
	TraceBatchStart(ctx context.Context, conn *Conn, data TraceBatchStartData) context.Context

	TraceBatchQuery(ctx context.Context, conn *Conn, data TraceBatchQueryData)
	TraceBatchEnd(ctx context.Context, conn *Conn, data TraceBatchEndData)
}
```
